airflow/api/common/trigger_dag.py

```diff
@@ -64,27 +64,31 @@ def _trigger_dag(
 
     if dag is None or dag_id not in dag_bag.dags:
         raise DagNotFound(f"Dag id {dag_id} not found")
-
-    logical_date = logical_date or timezone.utcnow()
-
-    if not timezone.is_localized(logical_date):
-        raise ValueError("The logical date should be localized")
-
-    if replace_microseconds:
-        logical_date = logical_date.replace(microsecond=0)
-
-    if dag.default_args and "start_date" in dag.default_args:
-        min_dag_start_date = dag.default_args["start_date"]
-        if min_dag_start_date and logical_date < min_dag_start_date:
-            raise ValueError(
-                f"Logical date [{logical_date.isoformat()}] should be >= start_date "
-                f"[{min_dag_start_date.isoformat()}] from DAG's default_args"
-            )
-    coerced_logical_date = timezone.coerce_datetime(logical_date)
-
-    data_interval = dag.timetable.infer_manual_data_interval(run_after=coerced_logical_date)
+    if logical_date:
+        if not timezone.is_localized(logical_date):
+            raise ValueError("The logical date should be localized")
+
+        if replace_microseconds:
+            logical_date = logical_date.replace(microsecond=0)
+
+        if dag.default_args and "start_date" in dag.default_args:
+            min_dag_start_date = dag.default_args["start_date"]
+            if min_dag_start_date and logical_date < min_dag_start_date:
+                raise ValueError(
+                    f"Logical date [{logical_date.isoformat()}] should be >= start_date "
+                    f"[{min_dag_start_date.isoformat()}] from DAG's default_args"
+                )
+        run_after = timezone.coerce_datetime(logical_date)
+        data_interval = dag.timetable.infer_manual_data_interval(run_after=run_after)
+    else:
+        data_interval = None
+        run_after = timezone.coerce_datetime(timezone.utcnow())
```

Comment:
```suggestion
        data_interval = dag.timetable.infer_manual_data_interval(run_after=run_after)
        run_after = timezone.coerce_datetime(logical_date)
    else:
        data_interval = None
        run_after = timezone.coerce_datetime(timezone.utcnow())
```

nit: reorder itt to keep the same reading order

Reply:
This is outdated change.